app/models/category.rb

```diff
@@ -18,6 +18,7 @@ class Category < ApplicationRecord
   def self.unarchived
     Category.
       joins(services: :location).
+      where(services: { archived_at: nil }).
       where(services: { locations: { archived_at: nil } }).
```

Comment:
These two lines can probably be combined

`where(services: { archived_at: nil, locations: { archived_at: nil } })`

Reply:
I think after the refactoring suggested by Dan, it will be good to go.👍🏽